### ISSUE TITLE
Call out unnecessary directives

### ISSUE THREAD

**Issue by matthewpersico** [koalaman/shellcheck#1656]
#### For bugs
- Rule Id (if any, e.g. SC1000): Any
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bash

# shellcheck disable=SC2230
w=$(which foo)

# shellcheck disable=SC2230
t=$(type -P foo)

print "$w $t"
```

#### Here's what shellcheck currently says:
```

```
#### Here's what I wanted or expected to see:

```
Line 6: SC9999 - Directive SC2330 does not apply.
```

I went through a round of replacing `which` with `type -P` and then I had to go back and get rid of all the SC2230 directives. It would be very useful if a directive that does not apply triggered an error so then you could more easily go back and get rid of them.
